feat: adds validation option on TextElement (#113)
Basis-Theory/developers.basistheory.com: docs/sdks/mobile/ios/options.mdx

  regexDigit
] as [Any]

let transformMatcher = try! NSRegularExpression(pattern: "[()-]") // Regex to remove parentheses & dashes

textElementUITextField.setConfig(options: TextElementOptions(mask: phoneMask, transform: ElementTransform(matcher: transformMatcher))
```

<Alert>
  If no transform is set, the value is sent to be tokenized as is, including the mask, if set.
</Alert>

## Validation

A [`TextElementUITextField`](/docs/sdks/mobile/ios/types#textelementuitextfield) allows you to validate user input before tokenization through the `validation` attribute of type `NSRegularExpression`.
Text is validated after every change and is considered valid if any matches are found using the `NSRegularExpression` provided. If the text is `nil` and a `validation` attribute is set, the text will be considered invalid. Let's run through

> an example. Say you want to validate a US based phone number. You can use the following `NSRegularExpression` to validate the phone number:

Reviewer: Should this text be closer to the text we have on the android side? https://developers.basistheory.com/docs/sdks/mobile/android/options#validator

```swift showLineNumbers
let regexDigit = try! NSRegularExpression(pattern: "\\d")

let phoneMask = [ // (123)456-7890
  "(",
  regexDigit,
  regexDigit,
  regexDigit,
  ")",
  regexDigit,
  regexDigit,
  regexDigit,
  "-",
  regexDigit,
  regexDigit,
  regexDigit,
  regexDigit
] as [Any]

let transformMatcher = try! NSRegularExpression(pattern: "[()-]") // Regex to remove parentheses & dashes
let phoneNumberRegex = try! NSRegularExpression(pattern: "^\\d{10}$") // Regex to validate phone number, after the transform is applied

textElementUITextField.setConfig(options: TextElementOptions(mask: phoneMask, transform: ElementTransform(matcher: transformMatcher), validation: phoneNumberRegex))
```

<Alert>

>   Validation is performed on the transformed value, if <a href="/docs/sdks/mobile/ios/options#transform">transform</a> is set. If I had not set a transform above to remove the parentheses and dashes, the validation would be performed on a

Reviewer: ```suggestion
  Validation is performed on the transformed value when <a href="/docs/sdks/mobile/ios/options#transform">transform</a> is set. Had a transform not been set above removing the parentheses and dashes, the validation would be performed on a
```
